app/services/credits/applied_coupon_service.rb

```diff
@@ -21,7 +21,9 @@ def create
         amount_currency: applied_coupon.amount_currency,
       )
 
-      applied_coupon.mark_as_terminated! if credit_amount >= remaining_amount
+      if applied_coupon.coupon.percentage? || credit_amount >= remaining_amount
```

Comment:
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`. (https://rubystyle.guide#if-as-a-modifier)